widget/gridwrap.go

```diff
@@ -54,6 +54,12 @@ type GridWrap struct {
 	// in the GridWrap has been unselected.
 	OnUnselected func(id GridWrapItemID) `json:"-"`
 
+	// OnHighlited is a callback to be notified when a given item
```

Comment:
Typo here

Reply:
will fix it. 

Should the current focus have a deprecated comment? or being it a private field is not necessary?

	// Deprecated: use currentHighlight instead
	currentFocus     ListItemID

Reply:
Great!!! 

Then I went ahead and: 

* Fixed the typo
* Renamed onFocuses to onHighlighted
* added similar callbacks for the mouse in event. 

As for the mouse events, I opted by having different callbacks for both keyboard and Mouse events, as this keeps them decoupled and gives options for users. Hope you are okay with it, otherwise please let me know what is it you prefer :)  


https://github.com/user-attachments/assets/b3e5669c-852b-4282-91b0-82917b42882b


OBS: The sample app I used in the video can be found on [coolapso/fyne-collection-highlight-hover-demo](https://github.com/coolapso/fyne-collection-highlight-hover-demo)

Reply:
I think I'd personally just have one callback to avoid potential confusion but I don't know what the other think

Reply:
Alright ... removed the OnHovered() callback and Mouse events now use the OnHighlighted() callback.